lib/Logger/LogAppenderSyslog.cpp

```diff
@@ -59,16 +96,13 @@ LogAppenderSyslog::LogAppenderSyslog(std::string const& facility,
   if ('0' <= facility[0] && facility[0] <= '9') {
     value = StringUtils::int32(facility);
   } else {
-    CODE* ptr = reinterpret_cast<CODE*>(facilitynames);
-
-    while (ptr->c_name != nullptr) {
-      if (strcmp(ptr->c_name, facility.c_str()) == 0) {
-        value = ptr->c_val;
-        break;
-      }
+    value = findSyslogFacilityByName(name);
+  }
 
-      ++ptr;
-    }
+  // Since the return value of ::openlog below is not checked or acted
+  // upon, try to be safe(er)
+  if (value < 0 or value > LOG_NFACILITIES) {
```

Comment:
Should this be `>=`? Wondering if there is a valid value at `facilitynames[LOG_NFACILITIES]`.
```suggestion
  if (value < 0 or value >= LOG_NFACILITIES) {
```